src/parquet/schema/descriptor.h

```diff
@@ -54,24 +53,32 @@ class ColumnDescriptor {
     return primitive_node_->physical_type();
   }
 
+  LogicalType::type logical_type() const {
+    return primitive_node_->logical_type();
+  }
+
   const std::string& name() const {
     return primitive_node_->name();
   }
 
   bool is_required() const {
-    return max_definition_level_ == 0;
+    return primitive_node_->is_required();
   }
 
   bool is_optional() const {
-    return max_definition_level_ > 0;
+    return primitive_node_->is_optional();
   }
 
   bool is_repeated() const {
-    return max_repetition_level_ > 0;
+    return primitive_node_->is_repeated();
   }
```

Comment:
These changes have fundamentally changed the nature of their results -- they can't be used like they were in the scanner any more, because you could have required types with definition level > 0:

```
optional group bag
  repeated group list
    required int32 item
```

I suggest using the max repetition/definition levels in the scanner to avoid this issue